Tests/Search/SearchSpec.swift

```diff
@@ -13,51 +13,51 @@ import Dispatch
-//                }
-//
-//                it("should return at least 1 result from locations with given type") {
-//                    sut.fetchData(keyword: "space")
-//                    await expect(sut.locationsWithGivenType.value.count).toEventually(beGreaterThanOrEqualTo(1), timeout: DispatchTimeInterval.seconds(10))
-//                }
-//
-//            }
-//
-//            context("when data fetching is called with search keyword 'adfjdskjhs'") {
-//
-//                it("should return 0 result from characters") {
-//                    sut.fetchData(keyword: "adfjdskjhs")
-//                    await expect(sut.characters.value.count).toEventually(equal(0), timeout: DispatchTimeInterval.seconds(10))
-//                }
-//
-//                it("should return 0 result from locations with given name") {
-//                    sut.fetchData(keyword: "adfjdskjhs")
-//                    await expect(sut.locatonsWithGivenName.value.count).toEventually(equal(0), timeout: DispatchTimeInterval.seconds(10))
-//                }
-//
-//                it("should return 0 result from locations with given name") {
-//                    sut.fetchData(keyword: "adfjdskjhs")
-//                    await expect(sut.locationsWithGivenType.value.count).toEventually(equal(0), timeout: DispatchTimeInterval.seconds(10))
-//                }
-//
-//            }
-//
-//        }
-//
-//    }
+    override func spec() {
+        describe("SearchViewModel") {
+
+            let sut = SearchViewModel()
+
+            context("when data fetching is called with search keyword 'space'") {
+
+                it("should return at least 1 result from characters") {
+                    sut.fetchData(input: "space")
+                    await expect(sut.characters.value.count).toEventually(beGreaterThanOrEqualTo(1), timeout: DispatchTimeInterval.seconds(10))
+                }
+
+                it("should return at least 1 result from locations with given name") {
+                    sut.fetchData(input: "space")
+                    await expect(sut.locatonsWithGivenName.value.count).toEventually(beGreaterThanOrEqualTo(1), timeout: DispatchTimeInterval.seconds(10))
+                }
+
+                it("should return at least 1 result from locations with given type") {
+                    sut.fetchData(input: "space")
+                    await expect(sut.locationsWithGivenType.value.count).toEventually(beGreaterThanOrEqualTo(1), timeout: DispatchTimeInterval.seconds(10))
+                }
+
+            }
+
+            context("when data fetching is called with search keyword 'adfjdskjhs'") {
+
+                it("should return 0 result from characters") {
+                    sut.fetchData(input: "adfjdskjhs")
+                    await expect(sut.characters.value.count).toEventually(equal(0), timeout: DispatchTimeInterval.seconds(10))
+                }
```

Comment:
I have a suggestion here:  insert a line of `sleep(1)` between `sut.fetchData()` and `await expect().toEventually()`. It's because I found that without `sleep(1)`, the test would always pass if it is expecting `0` as it was already `0` before the fetching.

Reply:
thanks just fixed